### PR TITLE
Upgrade to Bitblob v2.1.0 and use format specifiers

### PR DESCRIPTION
**Pull request by Geod24**

### CODE REVIEW

**Comment by Geod24** [comment]
Ping

File: source/agora/crypto/ECC.d [hunk 1]
```diff
@@ -129,17 +129,34 @@ public struct Scalar
 
     ***************************************************************************/
 
+    public void toString (scope void delegate(scope const(char)[]) @safe sink)
+        const
+    {
+        FormatSpec!char spec;
```
Comment (linked0) [inline source/agora/crypto/ECC.d:135]: In this case, what's the default format string?

File: source/agora/crypto/ECC.d [hunk 2]
```diff
@@ -129,17 +129,34 @@ public struct Scalar
...
 
-        case PrintMode.Clear:
-            formattedWrite(sink, "%s", this.data);
+        // Clear text was explicitly requested
+        case 'c':
+            this.data.toString(sink);
+            break;
+
+        // Modes supported by BitBlob
+        case 'x':
```
Comment (linked0) [inline source/agora/crypto/ECC.d:157]: Don't we need to add a test for this case?

**Review by linked0 — COMMENTED** [review]
Good but some comments.

File: source/agora/crypto/ECC.d (hunk 2 above)
Comment (Geod24) [reply source/agora/crypto/ECC.d:157]: The test is a L190

File: source/agora/crypto/ECC.d (hunk 1 above)
Comment (Geod24) [reply source/agora/crypto/ECC.d:135]: `'s'` is the default

File: source/agora/crypto/ECC.d (hunk 2 above)
Comment (linked0) [reply source/agora/crypto/ECC.d:157]: I understand what the code means. Ty.

**Review by linked0 — APPROVED** [review]
LGTM

**Review by kchulj — APPROVED** [review]
Nice improvement of the `toString`.

**Pull request merged**